lib/internal/errors.js

```diff
@@ -107,76 +105,81 @@ class SystemError extends Error {
       writable: true,
       configurable: true
     });
-    Object.defineProperty(this, kInfo, {
-      configurable: false,
-      enumerable: false,
-      value: context
-    });
-    Object.defineProperty(this, kCode, {
-      configurable: true,
-      enumerable: false,
-      value: key,
-      writable: true
-    });
     addCodeToName(this, 'SystemError', key);
-  }
 
-  get code() {
-    return this[kCode];
-  }
+    this.code = key;
 
-  set code(value) {
-    Object.defineProperty(this, 'code', {
-      configurable: true,
+    Object.defineProperty(this, 'info', {
+      value: context,
       enumerable: true,
-      value,
-      writable: true
+      configurable: true,
+      writable: false
     });
-  }
-
-  get info() {
-    return this[kInfo];
-  }
 
-  get errno() {
-    return this[kInfo].errno;
-  }
-
-  set errno(val) {
-    this[kInfo].errno = val;
-  }
-
-  get syscall() {
-    return this[kInfo].syscall;
-  }
-
-  set syscall(val) {
-    this[kInfo].syscall = val;
-  }
-
-  get path() {
-    return this[kInfo].path !== undefined ?
-      this[kInfo].path.toString() : undefined;
-  }
+    Object.defineProperty(this, 'errno', {
+      get() {
+        return context.errno;
+      },
+      set: (value) => {
+        context.errno = value;
+      },
+      enumerable: true,
+      configurable: true
+    });
 
-  set path(val) {
-    this[kInfo].path = val ?
-      lazyBuffer().from(val.toString()) : undefined;
-  }
+    Object.defineProperty(this, 'syscall', {
+      get() {
+        return context.syscall;
+      },
+      set: (value) => {
+        context.syscall = value;
+      },
+      enumerable: true,
+      configurable: true
+    });
 
-  get dest() {
-    return this[kInfo].path !== undefined ?
-      this[kInfo].dest.toString() : undefined;
-  }
+    if (context.path !== undefined) {
+      Object.defineProperty(this, 'path', {
+        get() {
+          return context.path != null ?
+            context.path.toString() : context.path;
+        },
+        set: (value) => {
+          context.path = value ?
+            lazyBuffer().from(value.toString()) : undefined;
```

Comment:
Given that `from()` can throw, users might be surprised by a throw when setting a property `error.path = 1` ... boom with an error about an invalid type passed to Buffer.from... may want to handle the error case a bit more explicitly here so the error is more directly relevant.

Reply:
I just kept it as it was before. I could add extra type validations in the setter but I personally would like to keep this separate.

Besides the fact that these setters are a bit weird in general: the `path` on `info` is normally a string. Using the setter it will be the buffer value of the input instead of the actually passed through value. I guess the reason for this is legacy support?

Reply:
I can see the reasoning. Perhaps just add some documentation around it?

Reply:
Done